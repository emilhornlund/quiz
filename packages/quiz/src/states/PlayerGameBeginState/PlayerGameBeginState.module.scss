@import '../../styles/colors';
@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  row-gap: $layout-gutter;
}
