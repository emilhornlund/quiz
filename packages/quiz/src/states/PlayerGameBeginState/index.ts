export type { PlayerGameBeginStateProps } from './PlayerGameBeginState'
export { default } from './PlayerGameBeginState'
