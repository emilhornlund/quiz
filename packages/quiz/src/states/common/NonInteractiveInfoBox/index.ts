export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
