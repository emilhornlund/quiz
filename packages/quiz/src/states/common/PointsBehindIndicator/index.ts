export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export { default } from './PointsBehindIndicator'
