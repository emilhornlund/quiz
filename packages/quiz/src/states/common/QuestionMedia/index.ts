export type { QuestionMediaProps } from './QuestionMedia'
export { default } from './QuestionMedia'
