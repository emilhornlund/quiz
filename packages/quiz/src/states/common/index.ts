export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
