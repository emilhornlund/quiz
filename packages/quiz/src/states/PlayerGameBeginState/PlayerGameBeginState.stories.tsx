import { GameEventType } from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'

import PlayerGameBeginState from './PlayerGameBeginState'

const meta = {
  component: PlayerGameBeginState,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof PlayerGameBeginState>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    event: {
      type: GameEventType.GameBeginPlayer,
      player: { nickname: 'FrostyBear' },
    },
  },
} satisfies Story
