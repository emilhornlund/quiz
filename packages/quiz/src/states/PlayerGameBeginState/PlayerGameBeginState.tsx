import { GameGameBeginPlayerEvent } from '@quiz/common'
import React, { FC } from 'react'

import {
  LoadingSpinner,
  NicknameChip,
  Page,
  Typography,
} from '../../components'

export interface PlayerGameBeginStateProps {
  event: GameGameBeginPlayerEvent
}

const PlayerGameBeginState: FC<PlayerGameBeginStateProps> = ({
  event: {
    player: { nickname },
  },
}) => (
  <Page>
    <Typography variant="title" size="medium">
      Get ready!
    </Typography>
    <Typography variant="text" size="small">
      The game starts any second
    </Typography>
    <NicknameChip value={nickname} />
    <LoadingSpinner />
  </Page>
)

export default PlayerGameBeginState
