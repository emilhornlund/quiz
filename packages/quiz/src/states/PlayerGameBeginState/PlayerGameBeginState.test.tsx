import { GameEventType } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import PlayerGameBeginState from './PlayerGameBeginState'

describe('PlayerGameBeginState', () => {
  it('should render PlayerGameBeginState', async () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerGameBeginState
          event={{
            type: GameEventType.GameBeginPlayer,
            player: { nickname: 'FrostyBear' },
          }}
        />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })
})
