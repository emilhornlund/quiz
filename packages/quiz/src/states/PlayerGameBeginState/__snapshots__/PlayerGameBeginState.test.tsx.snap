// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerGameBeginState > should render PlayerGameBeginState 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <span
          class="icon"
        />
        <span
          class="text"
        >
          Quiz
        </span>
      </button>
      <div
        class="side"
      />
    </div>
    <div
      class="content fullWidth centerAlign"
    >
      <div
        class="typography title medium"
      >
        Get ready!
      </div>
      <div
        class="typography text small"
      >
        The game starts any second
      </div>
      <div
        class="main"
      >
        FrostyBear
      </div>
      <div
        class="loadingSpinnerContainer"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
